diff -uNr musl-1.2.1.orig/include/paths.h musl-1.2.1/include/paths.h
--- musl-1.2.1.orig/include/paths.h	2020-08-03 23:21:09.000000000 -0500
+++ musl-1.2.1/include/paths.h	2020-12-14 21:50:07.613214612 -0600
@@ -18,9 +18,9 @@
 #define	_PATH_SHADOW	"/etc/shadow"
 #define	_PATH_SHELLS	"/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
-#define _PATH_UTMP	"/dev/null/utmp"
+#define _PATH_UTMP	"/run/utmps/utmp"
 #define	_PATH_VI	"/usr/bin/vi"
-#define _PATH_WTMP	"/dev/null/wtmp"
+#define _PATH_WTMP	"/run/utmps/wtmp"
 
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
