diff -uNr musl-1.2.2.orig/include/paths.h musl-1.2.2/include/paths.h
--- musl-1.2.2.orig/include/paths.h	2021-01-14 20:26:00.000000000 -0600
+++ musl-1.2.2/include/paths.h	2021-08-26 16:39:31.355618388 -0500
@@ -18,9 +18,9 @@
 #define	_PATH_SHADOW	"/etc/shadow"
 #define	_PATH_SHELLS	"/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
-#define _PATH_UTMP	"/dev/null/utmp"
+#define _PATH_UTMP	"/run/utmps/utmp"
 #define	_PATH_VI	"/usr/bin/vi"
-#define _PATH_WTMP	"/dev/null/wtmp"
+#define _PATH_WTMP	"/var/log/wtmp"
 
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
diff -uNr musl-1.2.2.orig/include/utmp.h musl-1.2.2/include/utmp.h
--- musl-1.2.2.orig/include/utmp.h	2021-01-14 20:26:00.000000000 -0600
+++ musl-1.2.2/include/utmp.h	2021-08-26 16:37:11.567365982 -0500
@@ -37,8 +37,8 @@
 
 int login_tty(int);
 
-#define _PATH_UTMP "/dev/null/utmp"
-#define _PATH_WTMP "/dev/null/wtmp"
+#define _PATH_UTMP "/run/utmps/utmp"
+#define _PATH_WTMP "/var/log/wtmp"
 
 #define UTMP_FILE _PATH_UTMP
 #define WTMP_FILE _PATH_WTMP
