--- e2fsprogs-1.45.5/e2fsck/unix.c.old	2020-01-06 23:10:17.000000000 +0000
+++ e2fsprogs-1.45.5/e2fsck/unix.c	2020-01-31 13:04:00.316309301 +0000
@@ -45,6 +45,7 @@
 #ifdef HAVE_DIRENT_H
 #include <dirent.h>
 #endif
+#include <inttypes.h>
 #include <libgen.h>
 
 #include "e2p/e2p.h"
@@ -1438,7 +1439,7 @@
 
 		fputs("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
 		      ctx->problem_logf);
-		fprintf(ctx->problem_logf, "<problem_log time=\"%lu\">\n",
+		fprintf(ctx->problem_logf, "<problem_log time=\"%"PRIu64"\">\n",
 			ctx->now);
 		fprintf(ctx->problem_logf, "<invocation prog=\"%s\"",
 			argv[0]);
