These tests do not work on PIE.  Disable them.

--- binutils-2.29/ld/testsuite/ld-elf/elf.exp.old	2017-08-19 22:39:31.236907813 +0000
+++ binutils-2.29/ld/testsuite/ld-elf/elf.exp	2017-08-20 00:12:04.735925829 +0000
@@ -296,27 +296,6 @@
      || [istarget *-*-nacl*]
      || [istarget *-*-gnu*] } {
     run_ld_link_exec_tests $array_tests_pie $xfails
-
-    run_ld_link_exec_tests [list \
-	[list \
-	    "Run mbind2a" \
-	    "$NOPIE_LDFLAGS -Wl,-z,common-page-size=0x4000" \
-	    "" \
-	    { mbind2a.s mbind2b.c } \
-	    "mbind2a" \
-	    "pass.out" \
-	    "-O2 -I../bfd" \
-	] \
-	[list \
-	    "Run mbind2b" \
-	    "-static -Wl,-z,common-page-size=0x4000" \
-	    "" \
-	    { mbind2a.s mbind2b.c } \
-	    "mbind2b" \
-	    "pass.out" \
-	    "-O2 -I../bfd" \
-	] \
-    ]
 }
 
 # <http://www.gnu.org/software/hurd/open_issues/binutils.html#static>
