--- binutils-2.31.1/ld/testsuite/ld-ifunc/ifunc.exp.old	2018-06-27 07:55:16.000000000 +0000
+++ binutils-2.31.1/ld/testsuite/ld-ifunc/ifunc.exp	2018-10-12 03:23:09.240000000 +0000
@@ -34,7 +34,8 @@
       && ([istarget "*-*-elf*"]
 	  || [istarget "*-*-nacl*"]
 	  || [istarget "*-*-linux*"]
-	  || [istarget "*-*-gnu*"])) } {
+	  || [istarget "*-*-gnu*"])
+      && ![istarget "*-*-musl*"]) } {
     verbose "IFUNC tests not run - target does not support IFUNC"
     return
 }
